L1Trigger/TrackFindingTracklet/plugins/ProducerDR.cc

```diff
@@ -0,0 +1,140 @@
+
+using namespace std;
+using namespace edm;
+using namespace trackerTFP;
+using namespace tt;
+
+namespace trklet {
+
+  /*! \class  trklet::ProducerDR
+   *  \brief  Emulates removal of duplicated TTTracks f/w
+   *  \author Thomas Schuh
+   *  \date   2023, Feb
+   */
+  class ProducerDR : public stream::EDProducer<> {
+  public:
+    explicit ProducerDR(const ParameterSet&);
+    ~ProducerDR() override {}
+
+  private:
+    void beginRun(const Run&, const EventSetup&) override;
+    void produce(Event&, const EventSetup&) override;
+    virtual void endJob() {}
+    // ED input token of Tracks
+    EDGetTokenT<StreamsTrack> edGetTokenTracks_;
+    // ED input token of Stubs
+    EDGetTokenT<StreamsStub> edGetTokenStubs_;
+    // ED output token for stubs
+    EDPutTokenT<StreamsStub> edPutTokenAcceptedStubs_;
+    EDPutTokenT<StreamsStub> edPutTokenLostStubs_;
+    // ED output token for tracks
+    EDPutTokenT<StreamsTrack> edPutTokenAcceptedTracks_;
+    EDPutTokenT<StreamsTrack> edPutTokenLostTracks_;
+    // Setup token
+    ESGetToken<Setup, SetupRcd> esGetTokenSetup_;
+    // DataFormats token
+    ESGetToken<DataFormats, DataFormatsRcd> esGetTokenDataFormats_;
+    // ChannelAssignment token
+    ESGetToken<ChannelAssignment, ChannelAssignmentRcd> esGetTokenChannelAssignment_;
+    // configuration
+    ParameterSet iConfig_;
+    // helper class to store configurations
+    const Setup* setup_ = nullptr;
+    // helper class to extract structured data from tt::Frames
+    const DataFormats* dataFormats_ = nullptr;
+    // helper class to assign tracks to channel
+    const ChannelAssignment* channelAssignment_ = nullptr;
+  };
+
+  ProducerDR::ProducerDR(const ParameterSet& iConfig) : iConfig_(iConfig) {
+    const string& label = iConfig.getParameter<string>("LabelDRin");
+    const string& branchAcceptedStubs = iConfig.getParameter<string>("BranchAcceptedStubs");
+    const string& branchAcceptedTracks = iConfig.getParameter<string>("BranchAcceptedTracks");
+    const string& branchLostStubs = iConfig.getParameter<string>("BranchLostStubs");
+    const string& branchLostTracks = iConfig.getParameter<string>("BranchLostTracks");
+    // book in- and output ED products
+    edGetTokenTracks_ = consumes<StreamsTrack>(InputTag(label, branchAcceptedTracks));
+    edGetTokenStubs_ = consumes<StreamsStub>(InputTag(label, branchAcceptedStubs));
+    edPutTokenAcceptedStubs_ = produces<StreamsStub>(branchAcceptedStubs);
+    edPutTokenAcceptedTracks_ = produces<StreamsTrack>(branchAcceptedTracks);
+    edPutTokenLostStubs_ = produces<StreamsStub>(branchLostStubs);
+    edPutTokenLostTracks_ = produces<StreamsTrack>(branchLostTracks);
+    // book ES products
+    esGetTokenSetup_ = esConsumes<Setup, SetupRcd, Transition::BeginRun>();
+    esGetTokenDataFormats_ = esConsumes<DataFormats, DataFormatsRcd, Transition::BeginRun>();
+    esGetTokenChannelAssignment_ = esConsumes<ChannelAssignment, ChannelAssignmentRcd, Transition::BeginRun>();
```

Comment:
I see two ways to use an initialization list:

1) To promote all the "const string" to class data members, so that they too can be set in an initialization list. This is ugly, since no code outside the class constructor accesses these "const strings", so they shouldn't really be data members.
2) To not explicitely define the "const strings" such as label, branchAcceptedTracks etc. at all, but instead to call iConfig.getParameter() on the fly whereever it is needed. The line of code https://github.com/cms-sw/cmssw/blob/master/L1Trigger/Phase2L1ParticleFlow/plugins/L1BJetProducer.cc#L44 illustrates this approach. It has the disadvantage of making the code harder to understand, and less CPU-efficient (since some of the const string are used a few times)